app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,16 @@
+class RecipesController < ApplicationController
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @ingredients = @recipe.ingredients
+    @total_cost = @recipe.total_cost
+  end
+
+  def create
+    @recipe = Recipe.find(params[:id])
```

Comment:
Nice work getting to the extensions! Your implementation looks good, but a more RESTful approach to this controller action would actually be to place it in a separate controller specifically for doing actions relating to the joins table. We call these joins controllers, and in this case you could have a RecipeIngredientsController with a `create` action that does exactly this. 

The problem with putting this implementation in the `recipes#create` action is that this action is typically reserved for creating Recipe resource, when in reality what you're doing here is creating an *association* between a single recipe and a single ingredient, i.e. creating a record in the joins table